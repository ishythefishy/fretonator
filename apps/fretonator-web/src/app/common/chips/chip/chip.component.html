<a
  [routerLink]="['/', note, noteExtender, mode]"
  title="note | createChipLink: noteExtender: mode"
  class="chip"
  [class.chip--active]="selected"
  [class.chip--round]="rounded"
  [class.chip--grey]="selectedColor === SelectedColor.grey"
>
  <ng-content></ng-content>
</a>
