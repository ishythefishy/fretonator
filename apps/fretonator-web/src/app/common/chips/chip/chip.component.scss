@import '../../../../styles/functions';
@import '../../../../styles/mixins';

:host {
  margin: pxToRem($grid-unit / 2);
  display: flex;
}

.chip {
  @include chip_button_base()
}

.chip--grey {
  border-color: var(--grey);
  color: var(--grey);
}

.chip--round {
  height: pxToRem(44);
  width: pxToRem(44);
  border-radius: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
}

.chip--active {
  background-color: var(--black);
  color: var(--white);

  &.chip--grey {
    background-color: var(--grey);
  }
}
