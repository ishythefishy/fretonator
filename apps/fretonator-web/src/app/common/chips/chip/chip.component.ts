import { Component, DoCheck, Input } from '@angular/core';
import { Mode } from '../../../util/types';
import { ActivatedRoute } from '@angular/router';

export enum SelectedColor {
  default = 'default',
  grey = 'grey',
}


@Component({
  selector: 'app-chip',
  templateUrl: './chip.component.html',
  styleUrls: ['./chip.component.scss']
})
export class ChipComponent implements DoCheck {
  @Input() note: string;
  @Input() noteExtender: string;
  @Input() mode: Mode;
  @Input() selectedColor: SelectedColor = SelectedColor.default;
  @Input() rounded = false;
  SelectedColor = SelectedColor;
  selected = false;

  constructor(private activatedRoute: ActivatedRoute) {
  }

  ngDoCheck() {
    this.selected = this.isSelected(this.activatedRoute);
  }

  isSelected(activatedRoute) {
    const data = activatedRoute.snapshot.data.selected;

    if (data.note !== this.note) {
      return false;
    }

    if (data.noteExtender !== this.noteExtender) {
      return false;
    }

    if (data.mode !== this.mode) {
      return false;
    }

    return true;
  }
}
