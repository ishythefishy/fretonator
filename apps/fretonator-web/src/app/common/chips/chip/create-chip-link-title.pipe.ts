import { Pipe, PipeTransform } from '@angular/core';
import { NoteExtenderString, Mode } from '../../../util/types';
import { GenerateFretMapService } from '../../generate-fret-map/generate-fret-map.service';

@Pipe({
  name: 'createChipLinkTitle',
})
export class CreateChipLinkTitlePipe implements PipeTransform {
  constructor(public generateFretMapService: GenerateFretMapService) {}
  transform(
    note: string,
    noteExtender: NoteExtenderString,
    mode: Mode
  ): string {
    return this.generateFretMapService.convertFretMapConfigurationToDisplayString(
      note,
      noteExtender,
      mode
    );
  }
}
