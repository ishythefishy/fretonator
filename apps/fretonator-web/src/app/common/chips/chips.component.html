<div class="chips">
  <ng-content select="app-chip"></ng-content>
</div>
