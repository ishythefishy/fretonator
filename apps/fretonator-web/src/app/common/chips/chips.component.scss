@import '../../../styles/functions';
@import '../../../styles/mixins';

.chips {
  padding-top: pxToRem($grid-unit);
  padding-bottom: pxToRem($grid-unit);
  display: flex;
  justify-content: flex-start;
  flex-wrap: wrap;
  margin: pxToRem($grid-unit * -1);

  @media screen and (min-width: 400px) {
    justify-content: center;
  }
}
