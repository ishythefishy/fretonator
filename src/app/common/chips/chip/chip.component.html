<input [id]="id"
       type="radio"
       class="input"
       [checked]="chips.value === value"
       [name]="chips.name"
       (change)="onChange()"
       (blur)="onBlur()"
       [disabled]="chips.disabled || disabled"
       [ngModel]="chips.value"
       [value]="value">

<label [for]="id" class="chip">
  <ng-content></ng-content>
</label>
