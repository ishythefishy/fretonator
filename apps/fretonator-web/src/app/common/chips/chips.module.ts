import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ChipsComponent } from './chips.component';
import { ChipComponent } from './chip/chip.component';
import { RouterModule } from '@angular/router';

@NgModule({
  declarations: [
    ChipsComponent,
    ChipComponent,
  ],
  imports: [
    CommonModule,
    RouterModule,
  ],
  exports: [
    ChipComponent,
    ChipsComponent,
  ],
})
export class ChipsModule {
}
