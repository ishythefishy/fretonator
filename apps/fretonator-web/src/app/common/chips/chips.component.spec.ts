import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { Component } from '@angular/core';
import { ChipsModule } from './chips.module';
import { ChipsComponent } from './chips.component';
import { SelectedColor } from './chip/chip.component';
import { RouterTestingModule } from '@angular/router/testing';
import { ActivatedRoute } from '@angular/router';

describe('ChipsComponent', () => {
  @Component({
    selector: 'app-chips-spec',
    template: `
      <app-chips>
        <app-chip
          [note]="'a'"
          [noteExtender]="'natural'"
          [mode]="'ionian'"
          [selectedColor]="selectedColor">One
        </app-chip>
        <app-chip
          [note]="'b'"
          [noteExtender]="'flat'"
          [mode]="'mixolydian'"
          [rounded]="true">Two
        </app-chip>
        <app-chip
          [note]="'f'"
          [noteExtender]="'sharp'"
          [mode]="'minorPentatonic'"
        >Two
        </app-chip>
      </app-chips>
    `
  })
  class ChipsComponentSpec {
    selectedColor: SelectedColor = SelectedColor.default;
  }

  let component: ChipsComponentSpec;
  let fixture: ComponentFixture<ChipsComponentSpec>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [
        ChipsComponentSpec
      ],
      imports: [
        ChipsModule,
        RouterTestingModule
      ],
      providers: [
        {
          provide: ActivatedRoute,
          useValue: {
            snapshot: {
              data: {
                selected: { note: 'c', noteExtender: 'natural', mode: 'ionian' }
              }
            }
          }
        }
      ]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ChipsComponentSpec);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match the snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
