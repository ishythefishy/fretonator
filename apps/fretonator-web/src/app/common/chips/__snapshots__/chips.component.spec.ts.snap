// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChipsComponent should match the snapshot 1`] = `
<app-chips-spec
  selectedColor={[Function String]}
>
  <app-chips>
    <div
      class="chips"
    >
      <app-chip>
        <a
          class="chip"
          href="/a/natural/ionian"
        >
          One 
        </a>
      </app-chip>
      <app-chip>
        <a
          class="chip chip--round"
          href="/b/flat/mixolydian"
        >
          Two 
        </a>
      </app-chip>
      <app-chip>
        <a
          class="chip"
          href="/f/sharp/minorPentatonic"
        >
          Two 
        </a>
      </app-chip>
    </div>
  </app-chips>
</app-chips-spec>
`;
